internal/controller/management/serviceaccount/serviceaccounts.go

```diff
@@ -0,0 +1,289 @@
+package serviceaccounts
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kelseyhightower/envconfig"
+	corev1 "k8s.io/api/core/v1"
+	kubeerr "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+const (
+	componentLabelKey    = "app.kubernetes.io/component"
+	controllerLabelValue = "controller"
+)
+
+type ReconcilerConfig struct {
+	KargoNamespace string `envconfig:"KARGO_NAMESPACE" required:"true"`
+}
+
+func ReconcilerConfigFromEnv() ReconcilerConfig {
+	cfg := ReconcilerConfig{}
+	envconfig.MustProcess("", &cfg)
+	return cfg
+}
+
+// reconciler reconciles ServiceAccount resources
+type reconciler struct {
+	cfg    ReconcilerConfig
+	client client.Client
+
+	getServiceAccountFn func(
+		context.Context,
+		types.NamespacedName,
+		client.Object,
+		...client.GetOption,
+	) error
+
+	createRoleBindingFn func(
+		context.Context,
+		client.Object,
+		...client.CreateOption,
+	) error
+
+	deleteRoleBindingFn func(
+		context.Context,
+		client.Object,
+		...client.DeleteOption,
+	) error
+
+	ensureControllerPermissionsFn func(context.Context, *corev1.ServiceAccount) error
+	removeControllerPermissionsFn func(context.Context, *corev1.ServiceAccount) error
+}
+
+// SetupReconcilerWithManager initializes a reconciler for ServiceAccount 
+// resources and registers it with the provided Manager.
+func SetupReconcilerWithManager(kargoMgr manager.Manager, cfg ReconcilerConfig) error {
+	return ctrl.NewControllerManagedBy(kargoMgr).
+		For(&corev1.ServiceAccount{}).
+		WithEventFilter(
+			predicate.Funcs{
+				CreateFunc: func(e event.CreateEvent) bool {
+					// Allow creation events to be handled for all ServiceAccounts
+					// This ensures any labeled or unlabeled SA gets proper
+					// reconciliation, including on restarts
+					return true
+				},
```

Comment:
iirc, if we don't define `CreateFunc` then no create events are filtered out. I think we only need to specify when there's a chance we do want to ignore something.